tests/test_delay.js

```diff
@@ -40,289 +38,277 @@ function checkDuration(start, durationMillis) {
   // .and.at.most(durationMillis + bufferMillis, 'delay upper bound exceeded')
 }
 
-test('calling delay could cause timeout error', async () => {
-  const scope = nock('http://example.test')
-    .get('/')
-    .delay({
-      head: 300,
-    })
-    .reply(200, 'OK')
-
-  await assertRejects(
-    got('http://example.test', { timeout: 100 }),
-    err => err.code === 'ETIMEDOUT'
-  )
-
-  scope.done()
-})
-
-test('Body delay does not have impact on timeout', async () => {
-  const scope = nock('http://example.test')
-    .get('/')
-    .delay({
-      head: 300,
-      body: 300,
-    })
-    .reply(201, 'OK')
+describe('`delay()`', () => {
+  let interceptor
+  let connSpy
+  let bodySpy
 
```

Comment:
This is an exception to our general strategy which is to test through the public API. Would it be worth adding a comment here explaining why, and that this should not be used as an example for writing new tests?

Reply:
Comment added.